### PR TITLE
fix: Disconnect db service on shutdown

### PR DESCRIPTION
**Pull request by BobdenOs**
Currently with `HANA` and `Postgres` the service does not disconnect with the default `cds.test`. So this registers the `disconnect` function to the `shutdown` event.

### CODE REVIEW

**Review by patricebender — APPROVED** [review]

File: db-service/lib/common/DatabaseService.js [hunk 1]
```diff
@@ -6,6 +6,12 @@ const cds = require('@sap/cds/lib')
 /** @typedef {unknown} DatabaseDriver */
 
 class DatabaseService extends cds.Service {
+
+  init() {
+    cds.on('shutdown', () => this.disconnect())
```
Comment (johannes-vogel) [inline db-service/lib/common/DatabaseService.js:11]: We might need to implement something that disconnects all tenant pools in multitenancy mode. Something like `disconnect('*')`

Comment (BobdenOs) [reply db-service/lib/common/DatabaseService.js:11]: `disconnect(undefined)` should match all tenants. Will update the `disconnect` function to just go over all pools.

File: db-service/lib/common/DatabaseService.js [hunk 2]
```diff
@@ -111,11 +117,18 @@ class DatabaseService extends cds.Service {
...
+      if (!pool) {
+        return
+      }
+      await pool.drain()
+      await pool.clear()
+      delete this.pools[tenant]
+    }
+    if (tenant == null)
+      return Promise.all(Object.keys(this.pools).map(_disconnect))
+    return _disconnect(tenant)
```
Comment (johannes-vogel) [inline db-service/lib/common/DatabaseService.js:131]:
```suggestion
      if (tenant == null)
        return Promise.all(Object.keys(this.pools).map(this.disconnect))
      const pool = this.pools[tenant]
      if (!pool) return
      await pool.drain()
      await pool.clear()
      delete this.pools[tenant]
    }
```

Comment (BobdenOs) [reply db-service/lib/common/DatabaseService.js:131]: It might be more understandable, but is also an infinite loop.

**Pull request merged**